posts/2024-03-29-ArgoCD-Configuration-Drift.adoc

```diff
@@ -0,0 +1,163 @@
+---
+layout: post
+title: "A Solution to Configuration Drift - Deploying Open Liberty with GitOps"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/danielguinan1
+author_github: https://github.com/danielguinan1
+seo-title: A Solution to Configuration Drift - Deploying Open Liberty with GitOps - OpenLiberty.io
+seo-description: Learn how to deploy an Open Liberty Application with GitOps
+blog_description: "Learn how to deploy an Open Liberty Application with GitOps"
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+additional_authors: 
+- name: Monica Tamboli
+  github: https://github.com/mtamboli
+  image: https://avatars0.githubusercontent.com/mtamboli
+---
+= A Solution to Configuration Drift - Deploying Open Liberty with GitOps 
+Daniel Guinan <https://github.com/danielguinan1>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+Many organizations are struggling with configuration drift, and GitOps provides a robust solution. If you're unfamiliar with what configuration drift is, how it can be caused, why it's bad or how GitOps can help to mitigate this, then check out part 1 of this two part blog series: [Configuration drift: Why its bad and how to solve it with GitOps and ArgoCD]
+
+In this blog, part 2 of this series, we will demonstrate how the solutions introduced in part 1 can be practically applied. We will walk you through the deployment of an Open Liberty application to a Kubernetes cluster using a continuous deployment approach with GitOps and ArgoCD (a leading GitOps tool designed to automate application deployment and ensure configuration consistency, effectively addressing the drift issue).
+
+* <<setup_argocd, See how to setup ArgoCD>>
+* <<setup_gitops_repo, Explore how to setup a GitOps repo for your app>>
+* <<deploy_app_with_gitops, Learn how to deploy a Java application with GitOps and ArgoCD>>
+* <<manually_sync, Discover how to manually sync your app>>
+* <<gitops_solve_config_drift, See how GitOps can solve Configuration Drift>>
+
+== Getting your Cluster Ready
+
+In this blog, we assume you have some form of kubernetes cluster, which you have access to using the `kubectl`CLI or `oc` CLI if on OpenShift. 
+
+=== Using the Open Liberty Operator
+
+The Open Liberty Operator streamlines the deployment and management of Open Liberty Applications on Kubernetes. Instead of manually crafting Kubernetes resources for the application (like services, deployments, routes, etc.), the Operator enables us to define a few application details, and it automatically generates the required Kubernetes resources. Furthermore, it simplifies day 2 operations such as traces and heap dumps when needed. For more on the Open Liberty Operator check out the docs on it https://openliberty.io/docs/latest/open-liberty-operator.html[here].
```

Comment:
```suggestion
The Open Liberty Operator streamlines the deployment and management of Open Liberty applications on Kubernetes. Instead of manually crafting Kubernetes resources for the application (like services, deployments, routes, etc.), the Operator only requires us to define a few application details, and it automatically generates the required Kubernetes resources - saving us time and effort. Furthermore, it simplifies day 2 operations such as traces and heap dumps when needed. For more on the Open Liberty Operator check out the docs on it https://openliberty.io/docs/latest/open-liberty-operator.html[here].
```